ProfilerGUI/Source/Configurator/MachineTypeExtractor.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.IO;
+
+namespace ProfilerGUI.Source.Configurator
+{
+    /// <summary>
+    /// Utilities to determine the machine type of a PE executable.
+    /// </summary>
+    internal class MachineTypeUtils
+    {
+        private const int ExpectedPEHead = 0x00004550;
+
+        /// <summary>
+        /// Tries to find the "bitness" (32bit vs 64bit) of the given executable by looking at the PE header.
+        /// This is needed to select the correct profiler DLL to run the application against.
+        ///
+        /// Taken from https://stackoverflow.com/questions/1001404/check-if-unmanaged-dll-is-32-bit-or-64-bit
```

Comment:
Isn't the article regarding unmanaged code?

Reply:
I tested this against a normal .NET executable and it selected the correct bitness

Reply:
I just tried the GUI with `test-data/test-programs/GeneratedTest.exe`, which is detected as 32bit. When I execute with this setting no trace is generated. When I manually switch to 64bit and run I get a trace file. I guess the detection fails, because `GeneratedTest.exe` is managed .NET code and bitness depends on the runtime environment, i.e., cannot be determined from the exe alone?

Reply:
Fair point, though in the case of a managed assembly we can still
- check if it has a fixed bitness (this is possible)
- if not, use the bitness of the machine (this is probably the correct bitness then, unless the user uses "prefer 32bit" in their project config explicitly, but then: whatever, let them deal with it ;-))

Reply:
Sounds fine to me :)